File: iosApp/iosApp/OnboardingComponents.swift
import SwiftUI

// ── Reusable onboarding UI components (mirror Android OnboardingComponents.kt) ─

// Hero icon (circle with person icon)
struct OnboardingHeroIcon: View {
    let colors: TexasWatchColors

    var body: some View {
        ZStack {
            Circle()
                .fill(colors.primaryAccent.opacity(0.12))
                .frame(width: 120, height: 120)
            Image(systemName: "person.2")
                .resizable()
                .scaledToFit()
                .frame(width: 52, height: 52)
                .foregroundColor(colors.primaryAccent)
        }
    }
}

// Header bar for full-text detail view
struct OnboardingHeaderBar<StartContent: View>: View {
    let title: String
    let colors: TexasWatchColors
    let typography: TexasWatchTypography
    @ViewBuilder let startContent: () -> StartContent

    var body: some View {
        HStack(spacing: 8) {
            startContent()
            Text(title)
                .font(typography.h3)
                .foregroundColor(colors.primaryText)
            Spacer()
        }
        .frame(height: 56)
        .padding(.horizontal, 4)
    }
}

// "Read more" action row
struct OnboardingActionRow: View {
    let label: String
    let colors: TexasWatchColors
    let typography: TexasWatchTypography
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack {
                Text(label)
                    .font(typography.text1)
                    .foregroundColor(colors.primaryText)
                Spacer()
                Image(systemName: "chevron.right")
                    .foregroundColor(colors.primaryAccent)
                    .frame(width: 20, height: 20)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(colors.mainBackground)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(colors.strokePale, lineWidth: 1)
            )
            .cornerRadius(12)
        }
        .buttonStyle(.plain)
    }
}

// Primary / secondary button
struct OnboardingButton: View {
    let label: String
    let primary: Bool
    let colors: TexasWatchColors
    let typography: TexasWatchTypography
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(typography.h4)
                .foregroundColor(primary ? colors.invertedText : colors.primaryText)
                .frame(maxWidth: .infinity)
                .frame(height: 48)
                .background(primary ? colors.primaryAccent : colors.mainBackground)
                .overlay(
                    RoundedRectangle(cornerRadius: 12)
                        .stroke(primary ? colors.primaryAccent : colors.strokePale, lineWidth: 1)
                )
                .cornerRadius(12)
        }
        .buttonStyle(.plain)
    }
}

// Toggle setting row
// onToggle is optional — if provided it intercepts the tap (used for permission requests).
// If nil, the binding is mutated directly.
struct OnboardingToggleItem: View {
    let title: String
    let description: String
    @Binding var isOn: Bool
    let colors: TexasWatchColors
    let typography: TexasWatchTypography
    var onToggle: ((Bool) -> Void)? = nil

    var body: some View {
        HStack(alignment: .center, spacing: 12) {
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(typography.h4)
                    .foregroundColor(colors.primaryText)
                Text(description)
                    .font(typography.text2)
                    .foregroundColor(colors.secondaryText)
            }
            Spacer()
            Toggle("", isOn: Binding(
                get: { isOn },
                set: { newValue in
                    if let onToggle {
                        onToggle(newValue)
                    } else {
                        isOn = newValue
                    }
                }
            ))
            .labelsHidden()
            .tint(colors.primaryAccent)
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(colors.strokePale, lineWidth: 1)
        )
        .cornerRadius(12)
    }
}

File: composeApp/src/androidMain/kotlin/com/jetbrains/spacetutorial/ui/OnboardingComponents.kt
package com.jetbrains.spacetutorial.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import com.jetbrains.spacetutorial.R
import com.jetbrains.spacetutorial.texaswatch.theme.TexasWatchTheme

// ── Hero icon placeholder (replace drawable with your own asset) ──────────────

@Composable
fun OnboardingHeroIcon(modifier: Modifier = Modifier) {
    val colors = TexasWatchTheme.colors
    Box(
        contentAlignment = Alignment.Center,
        modifier = modifier
            .size(120.dp)
            .clip(CircleShape)
            .background(colors.primaryAccent.copy(alpha = 0.12f)),
    ) {
        Icon(
            painter = painterResource(R.drawable.team_28),
            contentDescription = null,
            tint = colors.primaryAccent,
            modifier = Modifier.size(56.dp),
        )
    }
}

// ── Header bar used on the detail (full text) view ────────────────────────────

@Composable
fun OnboardingHeaderBar(
    title: String,
    startContent: @Composable () -> Unit = {},
) {
    val colors = TexasWatchTheme.colors
    val typography = TexasWatchTheme.typography
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .height(56.dp)
            .padding(horizontal = 4.dp),
    ) {
        startContent()
        Spacer(Modifier.width(8.dp))
        Text(
            text = title,
            style = typography.h3,
            color = colors.primaryText,
        )
    }
}

// ── "Read more" action row ─────────────────────────────────────────────────────

@Composable
fun OnboardingActionRow(
    label: String,
    onClick: () -> Unit,
) {
    val colors = TexasWatchTheme.colors
    val typography = TexasWatchTheme.typography
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .border(1.dp, colors.strokePale, RoundedCornerShape(12.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 16.dp, vertical = 14.dp),
    ) {
        Text(
            text = label,
            style = typography.text1,
            color = colors.primaryText,
        )
        Icon(
            painter = painterResource(R.drawable.arrow_left_24),
            contentDescription = null,
            tint = colors.primaryAccent,
            modifier = Modifier
                .size(20.dp)
                .scale(scaleX = -1f, scaleY = 1f),
        )
    }
}

// ── Primary / secondary button ────────────────────────────────────────────────

@Composable
fun OnboardingButton(
    label: String,
    onClick: () -> Unit,
    primary: Boolean,
    modifier: Modifier = Modifier,
) {
    val colors = TexasWatchTheme.colors
    val typography = TexasWatchTheme.typography
    val bgColor = if (primary) colors.primaryAccent else colors.mainBackground
    val textColor = if (primary) colors.invertedText else colors.primaryText
    val borderColor = if (primary) colors.primaryAccent else colors.strokePale

    Box(
        contentAlignment = Alignment.Center,
        modifier = modifier
            .height(48.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(bgColor)
            .border(1.dp, borderColor, RoundedCornerShape(12.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 16.dp),
    ) {
        Text(
            text = label,
            style = typography.h4,
            color = textColor,
        )
    }
}

// ── Toggle setting row ────────────────────────────────────────────────────────

@Composable
fun OnboardingToggleItem(
    title: String,
    description: String,
    enabled: Boolean,
    onToggle: (Boolean) -> Unit,
) {
    val colors = TexasWatchTheme.colors
    val typography = TexasWatchTheme.typography
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .border(1.dp, colors.strokePale, RoundedCornerShape(12.dp))
            .padding(horizontal = 16.dp, vertical = 12.dp),
    ) {
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                style = typography.h4,
                color = colors.primaryText,
            )
            Text(
                text = description,
                style = typography.text2,
                color = colors.secondaryText,
            )
        }
        Spacer(Modifier.width(12.dp))
        Switch(
            checked = enabled,
            onCheckedChange = onToggle,
            colors = SwitchDefaults.colors(
                checkedThumbColor = colors.invertedText,
                checkedTrackColor = colors.primaryAccent,
                uncheckedThumbColor = colors.secondaryText,
                uncheckedTrackColor = colors.strokePale,
            ),
        )
    }
}
